src/main/java/org/folio/service/upload/UploadDefinitionServiceImpl.java

```diff
@@ -378,6 +380,16 @@ private Errors validateFreeSpace(UploadDefinition definition, Errors errors) {
     }, tenantId);
   }
 
+  @Override
+  public Future<Boolean> updateUploadDefinitionStatusIfAllFilesUploadingFailed(UploadDefinition uploadDefinition, UploadDefinition.Status status, String tenantId) {
+    return uploadDefinition.getFileDefinitions()
+      .stream()
+      .allMatch(fileDefinition -> fileDefinition.getStatus().equals(ERROR))
+        ? uploadDefinitionDao.updateBlocking(uploadDefinition.getId(), definition ->
+          Future.succeededFuture(definition.withStatus(UploadDefinition.Status.ERROR)), tenantId)
+          .map(true)
+        : Future.succeededFuture(false);
+  }
```

Comment:
@RuslanLavrov Don't go crazy. I can't read it 😄 
I suggest the following
```suggestion
 public Future<Boolean> updateUploadDefinitionStatusIfAllFilesUploadingFailed(
          UploadDefinition uploadDefinition,
          Status status,
          String tenantId) {

    Future<Boolean> result = Future.succeededFuture(false);

    if (areAllDefinitionsFailed(uploadDefinition)) {
      result = uploadDefinitionDao
              .updateBlocking(uploadDefinition.getId(), it -> Future.succeededFuture(it.withStatus(Status.ERROR)), tenantId)
              .map(true);
    }

    return result;
  }


  private boolean areAllDefinitionsFailed(UploadDefinition uploadDefinition) {

    return uploadDefinition.getFileDefinitions().stream()
            .allMatch(it -> it.getStatus() == FileDefinition.Status.ERROR);
  }
```